app/services/permissions_checker.rb

```diff
@@ -56,4 +58,14 @@ def authorize_manage_recordings
 
     @current_user.recordings.find_by(record_id: @record_id).present?
   end
+
+  def authorize_room_limit
+    user = User.find(@user_id)
+    # return true if room limit has not been reached
+    if RolePermission.joins(:permission).find_by(role_id: user.role_id, permission: { name: 'RoomLimit' }).value.to_i <= user.rooms.count.to_i
+      return false
```

Comment:
This is not that crucial at this point,
But we can make a small refactoring to enable extracting the permissions hash and access the permission value here without making any extra queries to the database,
So, one query fetches the hash of `Hash(:permission_name => :value)` once and then the rest of the logic access it at the application level,